oauth2_http/javatests/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentialsTest.java

```diff
@@ -0,0 +1,821 @@
+            .setClientId(CLIENT_ID)
+            .setClientSecret(CLIENT_SECRET)
+            .setRefreshToken(REFRESH_TOKEN)
+            .setTokenUrl(TOKEN_URL)
+            .setTokenInfoUrl(TOKEN_INFO_URL)
+            .setRevokeUrl(REVOKE_URL)
+            .setAccessToken(new AccessToken(ACCESS_TOKEN, /* expirationTime= */ null))
+            .setQuotaProjectId(QUOTA_PROJECT)
+            .build();
+
+    assertEquals(credentials, secondCredentials);
+    assertEquals(secondCredentials, credentials);
+    assertEquals(credentials.hashCode(), secondCredentials.hashCode());
+  }
+
+  @Test
+  public void hashCode_differentCredentials() {
+    ExternalAccountAuthorizedUserCredentials credentials =
+        ExternalAccountAuthorizedUserCredentials.newBuilder()
+            .setAudience(AUDIENCE)
+            .setClientId(CLIENT_ID)
+            .setClientSecret(CLIENT_SECRET)
+            .setRefreshToken(REFRESH_TOKEN)
+            .setTokenUrl(TOKEN_URL)
+            .setTokenInfoUrl(TOKEN_INFO_URL)
+            .setRevokeUrl(REVOKE_URL)
+            .build();
+
+    // Second credentials have an access token set.
+    ExternalAccountAuthorizedUserCredentials secondCredentials =
+        ExternalAccountAuthorizedUserCredentials.newBuilder()
+            .setAudience(AUDIENCE)
+            .setClientId(CLIENT_ID)
+            .setClientSecret(CLIENT_SECRET)
+            .setRefreshToken(REFRESH_TOKEN)
+            .setTokenUrl(TOKEN_URL)
+            .setTokenInfoUrl(TOKEN_INFO_URL)
+            .setRevokeUrl(REVOKE_URL)
+            .setAccessToken(new AccessToken(ACCESS_TOKEN, /* expirationTime= */ null))
+            .setQuotaProjectId(QUOTA_PROJECT)
+            .build();
+
+    assertNotEquals(secondCredentials, credentials);
+    assertNotEquals(credentials, secondCredentials);
+    assertNotEquals(credentials.hashCode(), secondCredentials.hashCode());
+  }
+
+  @Test
+  public void hashCode_differentCredentialsWithCredentialsFile() throws IOException {
+    // Optional fields that can be specified in the credentials file.
+    List<String> fields = Arrays.asList("audience", "revoke_url", "quota_project_id");
+
+    // Credential initialized with all fields.
+    ExternalAccountAuthorizedUserCredentials credentials =
+        ExternalAccountAuthorizedUserCredentials.fromJson(buildJsonCredentials(), transportFactory);
+
+    for (String field : fields) {
+      // Build credential with one of these fields missing.
+      GenericJson json = buildJsonCredentials();
+      json.remove(field);
+
+      ExternalAccountAuthorizedUserCredentials secondCredentials =
+          ExternalAccountAuthorizedUserCredentials.fromJson(json, transportFactory);
+      assertNotEquals(secondCredentials, credentials);
+      assertNotEquals(credentials, secondCredentials);
+      assertNotEquals(credentials.hashCode(), secondCredentials.hashCode());
+    }
+  }
+
+  @Test
+  public void equals_differentCredentials() throws IOException {
```

Comment:
add equals tests for each property like in ServiceAccountCredential tests? Sorry for the churn, but should be relatively fast to copy-past them.